// zcatalyst-sdk-node/lib/utils/logger.js
"use strict";
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _Logger_instances, _Logger_getTimestamp, _Logger_logToConsole, _Logger_resetLogLevels;
Object.defineProperty(exports, "__esModule", { value: true });
exports.logger = exports.LEVEL = void 0;
const console_1 = require("console");
var LEVEL;
(function (LEVEL) {
    LEVEL["INFO"] = "info";
    LEVEL["WARN"] = "warn";
    LEVEL["ERROR"] = "error";
    LEVEL["DEBUG"] = "debug";
    LEVEL["ALL"] = "all";
    LEVEL["NONE"] = "none";
})(LEVEL = exports.LEVEL || (exports.LEVEL = {}));
class Logger {
    constructor(options) {
        _Logger_instances.add(this);
        this.logOptions = {
            enable_debug: (options === null || options === void 0 ? void 0 : options.enable_debug) || false,
            enable_error: (options === null || options === void 0 ? void 0 : options.enable_error) || false,
            enable_info: (options === null || options === void 0 ? void 0 : options.enable_info) || false,
            enable_warn: (options === null || options === void 0 ? void 0 : options.enable_warn) || false
        };
    }
    info(message) {
        if (this.logOptions.enable_info) {
            __classPrivateFieldGet(this, _Logger_instances, "m", _Logger_logToConsole).call(this, `[INFO] [${__classPrivateFieldGet(this, _Logger_instances, "m", _Logger_getTimestamp).call(this)}] : ${message}`);
        }
    }
    warn(message) {
        if (this.logOptions.enable_warn) {
            __classPrivateFieldGet(this, _Logger_instances, "m", _Logger_logToConsole).call(this, `[WARN] [${__classPrivateFieldGet(this, _Logger_instances, "m", _Logger_getTimestamp).call(this)}] : ${message}`);
        }
    }
    error(message) {
        if (this.logOptions.enable_error) {
            __classPrivateFieldGet(this, _Logger_instances, "m", _Logger_logToConsole).call(this, `[ERROR] [${__classPrivateFieldGet(this, _Logger_instances, "m", _Logger_getTimestamp).call(this)}] : ${message}`);
        }
    }
    debug(message) {
        if (this.logOptions.enable_debug) {
            __classPrivateFieldGet(this, _Logger_instances, "m", _Logger_logToConsole).call(this, `[DEBUG] [${__classPrivateFieldGet(this, _Logger_instances, "m", _Logger_getTimestamp).call(this)}] : ${message}`);
        }
    }
    setLogLevel(level = LEVEL.ALL) {
        // reset log levels
        __classPrivateFieldGet(this, _Logger_instances, "m", _Logger_resetLogLevels).call(this);
        switch (level) {
            case LEVEL.INFO:
                this.logOptions.enable_info = true;
                break;
            case LEVEL.WARN:
                this.logOptions.enable_warn = true;
                break;
            case LEVEL.ERROR:
                this.logOptions.enable_error = true;
                break;
            case LEVEL.DEBUG:
                this.logOptions.enable_debug = true;
                break;
            case LEVEL.ALL:
                this.logOptions = {
                    enable_debug: true,
                    enable_error: true,
                    enable_info: true,
                    enable_warn: true
                };
                break;
            case LEVEL.NONE:
                this.logOptions = {
                    enable_debug: false,
                    enable_error: false,
                    enable_info: false,
                    enable_warn: false
                };
                break;
        }
    }
}
_Logger_instances = new WeakSet(), _Logger_getTimestamp = function _Logger_getTimestamp() {
    return new Date().toISOString();
}, _Logger_logToConsole = function _Logger_logToConsole(message) {
    (0, console_1.log)(message);
}, _Logger_resetLogLevels = function _Logger_resetLogLevels() {
    this.logOptions = {
        enable_debug: false,
        enable_error: false,
        enable_info: false,
        enable_warn: false
    };
};
const logger = new Logger();
exports.logger = logger;
